### PR TITLE
[NO-JIRA] Move deprecated prop function to react-utils

### PR DESCRIPTION
**Pull request by georgegillams**

### CODE REVIEW

File: native/packages/react-native-bpk-component-text/src/common-types.js [hunk 1]
```diff
@@ -88,6 +89,7 @@ export const weightPropType = (
   componentName: string,
 ) => {
   if (!isWeightValid(props[propName], props.textStyle)) {
+    // eslint-disable-next-line no-console
```
Comment (georgegillams) [inline native/packages/react-native-bpk-component-text/src/common-types.js]: I don't know how CI was working without this before 🤷‍♂️

File: UNRELEASED.md [hunk 2]
```diff
@@ -1 +1,5 @@
 # Unreleased
+
+**Added:**
+ - bpk-react-utils:
+   - Added `depreciated` function that accepts a prop type and adds a suitable console warning when the prop is used.
```
Comment (shaundon) [inline UNRELEASED.md]:
```suggestion
   - Added `deprecated` function that accepts a prop type and adds a suitable console warning when the prop is used.
```

_Depreciation_ is when an asset loses value over time :trollface:

File: native/packages/react-native-bpk-component-text/src/common-types.js (hunk 1 above)
Comment (shaundon) [reply native/packages/react-native-bpk-component-text/src/common-types.js]: It only gives an eslint warning, not an error, I think. Nice one for spotting it though!

File: UNRELEASED.md (hunk 2 above)
Comment (georgegillams) [reply UNRELEASED.md]: I think in my previous life I must have been a banker, and I still have some old muscle memory 🤦‍♂️

**Pull request merged**